universal-application-tool-0.0.1/app/views/admin/programs/ProgramApplicationListView.java

```diff
@@ -0,0 +1,116 @@
+                h1("All Applications").withClasses(Styles.MY_4),
+                each(applications, this::renderApplicationListItem),
+                renderDownloadButton(programId));
+
+    return layout.render(head(layout.tailwindStyles()), body(contentDiv));
+  }
+
+  public Tag renderDownloadButton(long programId) {
+    String link = controllers.admin.routes.AdminApplicationController.downloadAll(programId).url();
+    return new LinkElement()
+        .setId("download-all-button")
+        .setHref(link)
+        .setText("Download all (CSV)")
+        .asButton();
+  }
+
+  public Tag renderApplicationListItem(Application application) {
+    String downloadLinkText = "Download (PDF)";
+    long applicationId = application.id;
+    String applicantName;
+    try {
+      String firstName =
+          application.getApplicantData().readString(WellKnownPaths.APPLICANT_FIRST_NAME).get();
+      String lastName =
+          application.getApplicantData().readString(WellKnownPaths.APPLICANT_LAST_NAME).get();
+      applicantName = String.format("%s, %s", lastName, firstName);
+    } catch (NoSuchElementException e) {
+      log.error("Application {} does not include an applicant name.");
+      applicantName = "<Anonymous Applicant>";
+    }
+    String lastEditText;
+    try {
+      lastEditText =
+          application
+              .getApplicantData()
+              .readString(WellKnownPaths.APPLICATION_SUBMITTED_TIME)
+              .get();
+    } catch (NoSuchElementException e) {
+      log.error("Application {} submitted without submission time marked.", applicationId);
+      lastEditText = "<ERROR>";
+    }
+    lastEditText = "Last edited " + lastEditText;
+
+    Tag topContent =
+        div(
+                div(
+                    div(applicantName)
+                        .withClasses(
+                            Styles.TEXT_BLACK, Styles.FONT_BOLD, Styles.TEXT_XL, Styles.MB_2)),
+                p().withClasses(Styles.FLEX_GROW))
+            .withClasses(Styles.FLEX);
+
+    Tag bottomContent =
+        div(
+                p(lastEditText).withClasses(Styles.TEXT_GRAY_700, Styles.ITALIC),
+                p().withClasses(Styles.FLEX_GROW),
+                renderDownloadLink(downloadLinkText, applicationId))
+            .withClasses(Styles.FLEX, Styles.TEXT_SM, Styles.W_FULL);
+
+    Tag innerDiv =
+        div(topContent, bottomContent)
+            .withClasses(
+                Styles.BORDER, Styles.BORDER_GRAY_300, Styles.BG_WHITE, Styles.ROUNDED, Styles.P_4);
+
+    return div(innerDiv).withClasses(Styles.W_FULL, Styles.SHADOW_LG, Styles.MB_4);
+  }
+
+  Tag renderDownloadLink(String text, long applicationId) {
+    String downloadLink =
+        controllers.admin.routes.AdminApplicationController.download(applicationId).url();
+
+    return new LinkElement()
+        .setId("application-download-link-" + applicationId)
+        .setHref(downloadLink)
+        .setText(text)
+        .setStyles(Styles.MR_2)
```

Comment:
Suggestion: (Non-Blocking) Consider adding `ReferenceClasses` here.

Reply:
Ah nice!  I have done this in the follow-on PR, assuming from "non-blocking" that that's okay.